src/anml/models/interface.py

```diff
@@ -78,3 +79,61 @@ def forward(self, x: np.ndarray, *args, **kwargs):
         # different from predict() in solver in the sense that both variable and data value can vary.
         # in predict() variable value is at taken to be the optimal.
         raise NotImplementedError()
+
+
+class TrimmingCompatibleModel(Model, ABC):
+
+    def __init__(self):
+        super().__init__()
```

Comment:
Maybe we could move `self._w` from concrete class to this abstract class :)

Reply:
Yes, good idea!